client/src/pages/platform/workflow-editor/components/WorkflowNodesPopoverMenu.tsx

```diff
@@ -132,15 +136,59 @@ const WorkflowNodesPopoverMenu = ({
     );
 
     const handleClusterElementClick = (data: ClusterElementDefinitionBasic) => {
-        if (agentData && setAgentData) {
-            let value = data.componentName;
+        if (!clusterElementsData || !sourceNode) return;
 
-            if (data.type === 'TOOLS') {
-                value = `${data.componentName}#${data.name}`;
-            }
+        const updatedClusterElementsData: ClusterElementsType = {
+            chatMemory: clusterElementsData.chatMemory,
+            model: clusterElementsData.model,
+            rag: clusterElementsData.rag,
+            tools: [...(clusterElementsData.tools || [])],
+        };
 
-            setAgentData({...agentData, [data.type]: value});
+        const propertyMap: Record<ClusterElementsDefinitionType, StoredClusterElementsType> = {
+            CHAT_MEMORY: 'chatMemory',
+            MODEL: 'model',
+            RAG: 'rag',
+        };
+
+        if (data.type === 'TOOLS') {
+            updatedClusterElementsData.tools = [
+                ...(clusterElementsData.tools || []),
+                {
+                    label: data.title,
+                    name: getFormattedClusterElementName(data.name, 'tools'),
+                    parameters: {},
+                    type: `${data.componentName}/v${data.componentVersion}/${data.name}`,
+                },
+            ];
+        } else {
+            if (data.type in propertyMap) {
+                updatedClusterElementsData[propertyMap[data.type as ClusterElementsDefinitionType]] = {
+                    label: data.title,
+                    name: getFormattedClusterElementName(
+                        data.componentName,
+                        propertyMap[data.type as ClusterElementsDefinitionType]
+                    ),
+                    parameters: {},
+                    type: `${data.componentName}/v${data.componentVersion}/${propertyMap[data.type as ClusterElementsDefinitionType]}`,
+                };
+            }
         }
+
+        setClusterElementsData?.(updatedClusterElementsData);
```

Comment:
nisan vidija ovaj pattern prije, jel ovo provjera ako `setClusterElementsData` postoji onda ga pozovi sa `updatedClusterElementsData` argumentom?

ako je ja bi to radije stavija u regularni if radi čitljivosti